File: flutter_bdmap/android/src/main/java/com/shingohu/flutter_bdmap/BDMapViewFactory.java
package com.shingohu.flutter_bdmap;

import android.content.Context;

import io.flutter.plugin.common.BinaryMessenger;
import io.flutter.plugin.common.StandardMessageCodec;
import io.flutter.plugin.platform.PlatformView;
import io.flutter.plugin.platform.PlatformViewFactory;

public class BDMapViewFactory extends PlatformViewFactory {

    private Context activityContext;
    private BinaryMessenger messenger;


    public BDMapViewFactory(Context activityContext, BinaryMessenger messenger) {
        super(StandardMessageCodec.INSTANCE);
        this.messenger = messenger;
        this.activityContext = activityContext;
    }

    @Override
    public PlatformView create(Context context, int viewId, Object args) {
        return new BDMapView(activityContext, messenger, viewId);
    }
}

File: flutter_bdmap/ios/Classes/BDMapViewFactory.swift
//
//  BDMapViewFactory.swift
//  flutter_bdmap
//
//  Created by 胡杰 on 2020/1/15.
//

import UIKit
import Flutter



public class BDMapViewFactory :NSObject, FlutterPlatformViewFactory{
    
    
    public func create(withFrame frame: CGRect, viewIdentifier viewId: Int64, arguments args: Any?) -> FlutterPlatformView {
        return BDMapView.init(withFrame: frame, viewIdentifier: viewId, arguments: args, binaryMessenger: _messenger!)
    }
    
    public func createArgsCodec() -> FlutterMessageCodec & NSObjectProtocol {
        return FlutterStandardMessageCodec.sharedInstance()
    }
    
    
    var _messenger : FlutterBinaryMessenger?
       
       init(messenger : FlutterBinaryMessenger) {
           super.init()
           
           self._messenger = messenger
           
           
       }
       
    
}
